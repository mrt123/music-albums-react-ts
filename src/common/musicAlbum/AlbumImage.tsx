interface AlbumImageProps {
  imageUrl: string | undefined;
}

const getCoverImageStyle = (imageUrl: string | undefined) => {
  return {
    width: "100px",
    minWidth: "100px",
    height: "100px",
    background: "blue",
    backgroundImage: imageUrl === undefined ? "none" : `url("${imageUrl}")`,
    backgroundSize: "contain",
    borderRadius: "50%",
    margin: "10px",
  };
};

const AlbumImage = ({ imageUrl }: AlbumImageProps) => {
  return (
    <div style={{ position: "relative" }}>
      <div style={getCoverImageStyle(imageUrl)}></div>
    </div>
  );
};

export default AlbumImage;
