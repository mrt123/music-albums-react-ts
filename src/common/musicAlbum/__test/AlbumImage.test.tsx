import { render } from "@testing-library/react";
import AlbumImage from "../AlbumImage";

describe("AlbumImage.tsx", () => {
  it("matches snapshot when imageUrl is undefined", () => {
    const { container } = render(<AlbumImage imageUrl={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
