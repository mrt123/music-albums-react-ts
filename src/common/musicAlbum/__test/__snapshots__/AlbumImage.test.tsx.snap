// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumImage.tsx matches snapshot when imageUrl is undefined 1`] = `
<div>
  <div
    style="position: relative;"
  >
    <div
      style="width: 100px; min-width: 100px; height: 100px; background: blue; background-image: none; background-size: contain; border-radius: 50%; margin: 10px;"
    />
    <div
      style="position: absolute; top: 0px; left: 0px;"
    />
  </div>
</div>
`;
